### ISSUE TITLE
hdr_rw location param is not associated to profiles when a DS is created

### ISSUE THREAD

**Issue by dneuman64** [apache/trafficcontrol#2918]
If you create a new delivery service that has a edge header rewrite Traffic Ops correctly creates the location parameter, but it is not assigned to any profiles.   This means that if you don't manually associate the parameter with your Edge cache profiles then ORT will fail.  This used to work in the perl Traffic Ops code.

**Comment by rob05c** [comment]
TLDR Go is correctly emulating the Perl API, but the real bug is that the Perl API never did what the Perl UI does.

It looks like there's a mismatch between the UI and API. The Perl UI creates and assigns various location parameters, but in `deliveryservice_servers`, not `deliveryservice`. Which makes sense: when the DS is first created, there are no servers assigned, so we can't know which Server Profiles to assign them to.

So, unless there are objections, my plan is to update the Go API to match the Perl UI, and create and assign all location params as necessary, in the `POST+PUT deliveryservice_server` endpoints rather than the `POST deliveryservice`.

That should make the Portal behave the same way the Perl UI does.

**Comment by rob05c** [comment]
Related: https://github.com/apache/trafficcontrol/issues/2919
